src/Fixer/Whitespace/TypesSpacesFixer.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class TypesSpacesFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'A single space or none should be around union type operator.',
+            [
+                new CodeSample(
+                    "<?php\ntry\n{\n    new Foo();\n} catch (ErrorA | ErrorB \$e) {\necho'error';}\n"
+                ),
+                new CodeSample(
+                    "<?php\ntry\n{\n    new Foo();\n} catch (ErrorA|ErrorB \$e) {\necho'error';}\n",
+                    ['space' => 'single']
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction foo(int | string \$x)\n{\n}\n",
+                    new VersionSpecification(80000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(CT::T_TYPE_ALTERNATION);
+    }
+
+    public function getPriority(): int
+    {
+        return -33; // TODO: remove once https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5738 is merged
```

Comment:
#5738 is merged

Reply:
@jrmajor yes, it is, and this PR is up to date and ready to review (failing build is fixed in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5789).

Reply:
I'm not sure what #5789 has to do with it... What I understood from this comment is that priority can be removed, when #5738 is merged.

Reply:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5789 has to do with it exactly what I wrote - it's fixing failing build you can see here.

I've removed this `getPriority` function.